src/VaultSharp/V1/SecretsEngines/PKI/PKIRole.cs

```diff
@@ -0,0 +1,322 @@
+﻿using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace VaultSharp.V1.SecretsEngines.PKI
+{
+    public class PKIRole
+    {
+        /// <summary>
+        /// If set, clients can request certificates for any domain, regardless of allowed_domains restrictions. See the documentation for more information.
+        /// </summary>
+        [JsonPropertyName("allow_any_name")]
+        public bool AllowAnyName { get; set; }
+
+        /// <summary>
+        /// If set, clients can request certificates for the base domains themselves, e.g. "example.com" of domains listed in allowed_domains. 
+        /// This is a separate option as in some cases this can be considered a security threat. See the documentation for more information.
+        /// </summary>
+        [JsonPropertyName("allow_bare_domains")]
+        public bool AllowBareDomains { get; set; }
+
+        /// <summary>
+        /// If set, domains specified in allowed_domains can include shell-style glob patterns, e.g. "ftp*.example.com". 
+        /// See the documentation for more information.
+        /// </summary>
+        [JsonPropertyName("allow_glob_domains")]
+        public bool AllowGlobDomains { get; set; }
+
+        /// <summary>
+        /// If set, IP Subject Alternative Names are allowed. Any valid IP is accepted and No authorization checking is performed.
+        /// </summary>
+        [JsonPropertyName("allow_ip_sans")]
+        public bool AllowIpSans { get; set; }
```

Comment:
The default value of this flag is `true`. Wondering, if we should set the default values for booleans here, to prevent unintended configurations. 

Need to review other such flags within the request. 

https://developer.hashicorp.com/vault/api-docs/secret/pki#create-update-role